proto/src/main/proto/grpc.proto

```diff
@@ -598,6 +598,145 @@ message AddressBalanceInfo {
     bool is_address_unused = 4;
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////
+// MoneroConnections
+///////////////////////////////////////////////////////////////////////////////////////////
+
+service MoneroConnections {
+    rpc AddConnection (AddConnectionRequest) returns (AddConnectionReply) {
+    }
+
+    rpc RemoveConnection(RemoveConnectionRequest) returns (RemoveConnectionReply) {
+    }
+
+    rpc GetConnection(GetConnectionRequest) returns (GetConnectionReply) {
+    }
+
+    rpc GetConnections(GetConnectionsRequest) returns (GetConnectionsReply) {
+    }
+
+    rpc SetConnection(SetConnectionRequest) returns (SetConnectionReply) {
+    }
+
+    rpc ExtendedSetConnection(ExtendedSetConnectionRequest) returns (ExtendedSetConnectionReply) {
+    }
+
+    rpc CheckCurrentConnection(CheckCurrentConnectionRequest) returns (CheckCurrentConnectionReply) {
+    }
+
+    rpc CheckConnection(CheckConnectionRequest) returns (CheckConnectionReply) {
+    }
+
+    rpc CheckConnections(CheckConnectionsRequest) returns (CheckConnectionsReply) {
+    }
+
+    rpc StartCheckingConnections(StartCheckingConnectionsRequest) returns (StartCheckingConnectionsReply) {
+    }
+
+    rpc StopCheckingConnections(StopCheckingConnectionsRequest) returns (StopCheckingConnectionsReply) {
+    }
+
+    rpc GetBestAvailableConnection(GetBestAvailableConnectionRequest) returns (GetBestAvailableConnectionReply) {
+    }
+
+    rpc SetAutoSwitch(SetAutoSwitchRequest) returns (SetAutoSwitchReply) {
+    }
+
+}
+
+
+message AddConnectionRequest {
+    UriConnection connection = 1;
+}
+
+message AddConnectionReply {}
+
+message RemoveConnectionRequest {
+    string uri = 1;
+}
+
+message RemoveConnectionReply {}
+
+message GetConnectionRequest {}
+
+message GetConnectionReply {
+    UriConnection connection = 1;
+}
+
+message GetConnectionsRequest {}
+
+message GetConnectionsReply {
+    repeated UriConnection connections = 1;
+}
+
+message SetConnectionRequest {
+    string uri = 1;
+}
+
+message SetConnectionReply {}
+
+message ExtendedSetConnectionRequest {
+    UriConnection connection = 1;
+}
+
+message ExtendedSetConnectionReply {}
+
+message CheckCurrentConnectionRequest {}
+
+message CheckCurrentConnectionReply {
+    UriConnection connection = 1;
+}
+
+message CheckConnectionRequest {
+    UriConnection connection = 1;
+}
+
+message CheckConnectionReply {
+    UriConnection connection = 1;
+}
+
+message CheckConnectionsRequest {}
+
+message CheckConnectionsReply {
+    repeated UriConnection connections = 1;
+}
+
+message StartCheckingConnectionsRequest {
+    int32 refreshPeriod = 1; // milliseconds
```

Comment:
Use underscores, not camelcase: `refresh_period`. Otherwise the camelcase is incorrect for the grpc client stubs.